### PR TITLE
feat: performance gate

### PR DESCRIPTION
**Pull request by NDStrahilevitz**
### 1. Explain what the PR does

"Replace me with `make check-pr` output"

### CODE REVIEW

**Comment by geyslan** [comment]
@NDStrahilevitz is this error occurring since it's not merged yet?

https://github.com/aquasecurity/tracee/actions/runs/8589002210

```
Error
every step must define a `uses` or `run` key
```

**Comment by NDStrahilevitz** [comment]
> @NDStrahilevitz is this error occurring since it's not merged yet?
> 
> https://github.com/aquasecurity/tracee/actions/runs/8589002210
> 
> ```
> Error
> every step must define a `uses` or `run` key
> ```

I'm pretty sure this is from an old version of the PR, as checks are green currently. And no, it was a syntax error.

**Review by geyslan — APPROVED** [review]
LGTM. Put some thoughts.

File: .github/workflows/performance.yaml
Comment (NDStrahilevitz) [reply .github/workflows/performance.yaml:77]: @aquasecurity/tracee created this now so that we can.

File: .github/workflows/performance.yaml
Comment (NDStrahilevitz) [reply .github/workflows/performance.yaml:79]: If our last latency ratio was 1.4 and after a PR it is 2.9, there will be an alert. But if its 1.6 there won't be. We may want to make this percent smaller to avoid drift.

File: performance/benchmark/network/bench.sh
Comment (NDStrahilevitz) [reply performance/benchmark/network/bench.sh:5]: Yes, otherwise we might have this leftover in the cluster.

**Comment by NDStrahilevitz** [comment]
@geyslan I've pinned all the imported actions to a commit like you suggested. Thanks for alerting me on that.

**Comment by geyslan** [comment]
https://github.com/aquasecurity/tracee/actions/runs/8771893344/job/24070221960?pr=3945

I've been dealing with this issue for a while now... it seems like other apt locks are being triggered (and not properly released during the life of the script). The last solution I will bring up will be calling `wait_for_apt_locks()` before all apt-get calls.

**Comment by NDStrahilevitz** [comment]
Still waiting on @josedonizetti to review before merging.

**Comment by geyslan** [comment]
As an exception this can be cherry-picked into v0.21.0 RC if it's all ok after @josedonizetti review. If it is not possible to have it in time, please change the milestone label.

**Comment by NDStrahilevitz** [comment]
> As an exception this can be cherry-picked into v0.21.0 RC if it's all ok after @josedonizetti review. If it is not possible to have it in time, please change the milestone label.

I wouldn't say its necessary to cherry-pick, as the trigger is only on main anyway. I'll move the milestone. I still would like this to be reviewed and merged ASAP ofc.

**Review by josedonizetti — APPROVED** [review]
amazing work @NDStrahilevitz

**Comment by NDStrahilevitz** [comment]
Merged, now to see if it actually works as expected 🙃

**Comment by NDStrahilevitz** [comment]
Looks like the infrastructure broke (runner not picking up the job): https://github.com/aquasecurity/tracee/settings/actions/runners/34385

**Pull request merged**
